include/deal.II/grid/grid_generator.h

```diff
@@ -60,6 +61,173 @@ namespace GridGenerator
    */
   ///@{
 
+  /**
+   * TODO
+   *
+   * @ingroup GridGenerator
+   */
+  namespace Airfoil
+  {
+    /**
+     * AdditionalData collects all settings that are required to generate a
+     * airfoil triangulation with the functions Airfoil::create_triangulation().
+     */
+    struct AdditionalData
+    {
+      /**
+       * Either "NACA" or "Jouk" to choose airfoil geometry among NACA and
+       * Joukowski airfoil.
+       */
+      std::string airfoil_type;
+
+      /**
+       * NACA serial number defining the airfoil shape.
+       */
+      std::string naca_id;
+
+      /**
+       * Center of Joukowski circle.
+       *
+       * @note A center on the x-axis leads to a symmetrical airfoil.
+       */
+      Point<2, double> center;
+
+      /**
+       * Chord length of the airfoil, i.e. distance from leading to trailing
+       * edge.
+       */
+      double airfoil_length;
+
+
+
+      /**
+       * Vertical distance from airfoil chord to upper boundary of the mesh
+       * i.e. half of the total mesh hight.
+       */
+      double height;
+
+      /**
+       * Length of mesh from the airfoil trailing edge to outflow boundary.
+       */
+      double length_b2;
+
+
+
+      /**
+       * Factor defining the inclination of the face that block 2 and 3
+       * or block 5 and 6 share.
+       */
+      double incline_factor;
+
+      /**
+       * Factor to receive a finer mesh around the airfoil by increasing it.
+       */
+      double bias_factor;
+
+      /**
+       * Global refinement-level.
+       */
+      int refinements;
+
+      /**
+       * Number of subdivisions along the airfoil in left block.
+       */
+      int n_subdivision_x_0;
+
+      /**
+       * Number of subdivisions along the airfoil in middle block.
+       */
+      int n_subdivision_x_1;
+
+      /**
+       * Number of subdivisions in block right of the airfoil.
+       */
+      int n_subdivision_x_2;
+
+      /**
+       * Number of subdivisions normal to the airfoil contour.
+       */
+      int n_subdivision_y;
+
+
+      /**
+       * TODO
+       */
+      unsigned int point_factor;
+
+      /**
+       * Constructor.
+       */
+      AdditionalData();
+
+      /**
+       * This Function declares the ParameterHandler entries.
+       *
+       * @param prm Parameter handler.
+       */
+      void
+      declaration(ParameterHandler &prm);
+
+      /**
+       * This Function gets the entry values out of the
+       * ParameterHandler object and assigns those to the corresponding
+       * parameters in class AdditionalData.
+       *
+       * @param prm Parameter handler.
+       */
+      void
+      get(ParameterHandler &prm);
```

Comment:
Is there a reason not to use the `add_parameter` feature of `ParameterHandler`? This would allow you to attach a parameter right at the time of the declaration, and would remove the above `get` function (simplifying the code, and removing the possibility of bugs when adding/removing parameters in future versions).

Reply:
> Nice work! I only know for the NACA0012 airfoil, but for that the surface is given as an analytically known function that should be relatively easy to encode in a manifold. One would then use the `TransfiniteInterpolationManifold` class for the remainder of the mesh.
> 
> @eliasstudiert -- there are a number of TODOs or unfinished sentences still left in the documentation in the .h file. If you wanted to go through those first, we can save one round of reviews and the whole process is likely going to be a bit faster than if we have to mark all of these places up. Can you go through your text again and let us know when these are addressed?

I added the documentation and finished the TODOs. Also I simplified the function interpolate() in taking advantage of symmetry of the coarse grid. The code is ready to review. I didn't figure out so far, why `create_triangulation()` without `AdditionalData data` as inputparameter (default setup) doesn't work without providing the inputfile.